src/Tools/dotnet-user-jwts/src/Helpers/DevJwtCliHelpers.cs

```diff
@@ -0,0 +1,166 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text.Json;
+using System.Xml.Linq;
+using System.Xml.XPath;
+using Microsoft.Extensions.CommandLineUtils;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Configuration.UserSecrets;
+
+namespace Microsoft.AspNetCore.Authentication.JwtBearer.Tools;
+
+internal static class DevJwtCliHelpers
+{
+    public static string GetUserSecretsId(string projectFilePath)
+    {
+        var projectDocument = XDocument.Load(projectFilePath, LoadOptions.PreserveWhitespace);
+        var existingUserSecretsId = projectDocument.XPathSelectElements("//UserSecretsId").FirstOrDefault();
+
+        if (existingUserSecretsId == null)
+        {
+            return null;
+        }
+
+        return existingUserSecretsId.Value;
+    }
+
+    public static string GetProject(string projectPath = null)
+    {
+        if (projectPath is not null)
+        {
+            return projectPath;
+        }
+
+        var csprojFiles = Directory.GetFiles(Directory.GetCurrentDirectory(), "*.csproj");
```

Comment:
This should support *.vbproj and *.fsproj as well I think. I believe `dotnet user-secrets` already does.

Reply:
I think we should be reusing dotnet-user-secrets ProjectIdResolver. https://github.com/dotnet/aspnetcore/blob/2d2295183b26b98c386d31c29403328e9c1f53ef/src/Tools/dotnet-user-secrets/src/Internal/ProjectIdResolver.cs

Reply:
I think we really only need this bit.

```csharp
var csprojFiles = Directory.EnumerateFileSystemEntries(Directory.GetCurrentDirectory(), "*.*proj", SearchOption.TopDirectoryOnly)
                .Where(f => !".xproj".Equals(Path.GetExtension(f), StringComparison.OrdinalIgnoreCase))
                .ToList();
```

Reply:
`".xproj"`! 

![image](https://th.bing.com/th/id/R.2605083c31156e7b83126174ef8c0685?rik=SB4i7iPWZqfd0Q&riu=http%3a%2f%2fmedia.giphy.com%2fmedia%2fVMgcrwq9imGHu%2fgiphy.gif&ehk=%2bG3NWUmL6OOJxXice%2b3aiFL4FWqpaPjDlPChWyuKP14%3d&risl=&pid=ImgRaw&r=0)

Reply:
I think we need the whole thing. Using msbuild to run [SecretManager.targets](https://github.com/dotnet/aspnetcore/blob/2d2295183b26b98c386d31c29403328e9c1f53ef/src/Tools/dotnet-user-secrets/src/assets/SecretManager.targets) definitely seems like the most reliable way to do this for all project configurations. If `dotnet user-secrets` does this, I don't know why `dotnet user-jwts` wouldn't.

We can factor out `IReporter` and stuff if we need to, but I don't think we do. Error reporting is nice.

Reply:
Yeah to @halter73's point, folks could be specifying this property in a different MSBuild file so it really should be discovered via MSBuild. We can just log it as an issue and address it in a later preview.

Reply:
Filed into https://github.com/dotnet/aspnetcore/issues/41888.